crates/indexer/src/processors/default_processor.rs

```diff
@@ -524,12 +526,54 @@ impl TransactionProcessor for DefaultTransactionProcessor {
         }
 
         // TODO, merge this loop with above
+        // Moving object handling here because we need a single object
```

Comment:
What's the difference between this new way and the old way of getting the objects? They look like they're doing the same thing.

Reply:
They are. If you notice though that I'm passing all_objects into the from_delete_resource function. That's the main difference. I mean I could've made it work as before as well but the right thing to do is to use a general forloop to avoid the duplicate work so I just refactored this at the same time.